x-pack/test_utils/README.md

```diff
@@ -98,10 +98,50 @@ type TestSubjects = 'indicesTable' | 'createIndexButton' | 'pageTitle';
 export const setup = registerTestBed<TestSubjects>(MyComponent);
 ```
 
+## Best practices
+
+In order to prevent flakiness in component integration tests, please consider the following best practices: 
+
+- **Use** `jest.useFakeTimers()`.
+  
+  The code under a test might be using `setTimeout()` calls. This is bad for deterministic tests. In order to avoid it, we need to use mocked timers from jest. For that we declare at the top of each component integration test the following:
+​
+  ```js
+  beforeAll(() => {
+    jest.useFakeTimers();
+  });
+  ​
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+  ```
+
+- **Do not use** using `nextTick()`, `waitFor`, or `waitForFunc` helpers in tests. These helpers use `setTimeout` underneath and add latency in the tests, especially on CI where a timeout (even of a few ms) can trigger a timeout error. These helpers will eventually be deprecated once existing tests has been updated.
```

Comment:
Would there be any downside to adding `console.warn` calls to these functions? This will help people developing tests realize they shouldn't use these helpers. They'll emit in CI too but I don't see a downside to that.

Reply:
I thought about this as well. The one downside I can think of is `nextTick` is used a fair bit in some of the existing tests, so I could see it being a little noisy. I did a quick search and I don't see `waitForFunc` being used anywhere, so I think this one can be safely removed. `waitFor` is only used in remote clusters and CCR right now.

Reply:
I think our team is the only consumer of these functions so I thought that being all warned 😊 removes the need for a console.warn. The idea is to deprecate and remove those functions eventually.